### ISSUE TITLE
Add `_ :: Some1 (f1 :: k1 -> k0) -> DemoteRep ('KProxy :: KProxy k1)` and friends.

### ISSUE THREAD

**Issue by k0001** [k0001/exinst#3]
``` haskell
foo :: Some1 (f1 :: k1 -> k0) -> DemoteRep ('KProxy :: KProxy k1)
foo = \(Some1 a1 _) -> fromSing a1
```

And similar for `Some{2,3,4}`. 

The biggest issue with this is that I can't think of a nice name for this function :)

**Comment by k0001** [comment]
cc @isovector

**Comment by isovector** [comment]
How about `domImage`, since it gives the image under the domain of the functor `f1 :: k1 -> k0`? Not necessarily catchy, but at least it's informative :)

I'd suggest something along the lines of `_Projection` for the codomain prisms.

**Comment by k0001** [comment]
@isovector I'm still not convinced this deserves to be added. Does it? Maybe like this:

``` haskell

some1singRep :: Some1 (f1 :: k1 -> k0) -> DemoteRep k1
some2singRep :: Some2 (f2 :: k2 -> k1 -> k0) -> (DemoteRep k2, DemoteRep k1)
some3singRep :: Some3 (f3 :: k3 -> k2 -> k1 -> k0) -> (DemoteRep k3, DemoteRep k2, DemoteRep k1)
some4singRep :: Some4 (f4 :: k4 -> k3 -> k2 -> k1 -> k0) -> (DemoteRep k4, DemoteRep k3, DemoteRep k2, DemoteRep k1)
```

I'm not convinced really, maybe we should just live this functionality out since it is so straightforward to implement, anyway.

**Comment by isovector** [comment]
I think it does; `SomeN` is essentially a dependent tuple, and projecting bits of a tuple is the only thing you ever want to do with them. I ran into needing `some1singRep` within 20 minutes of using the library, and don't consider my use-case to be particularly weird.

**Comment by k0001** [comment]
Fixed in d191ec288f6bc3fae27cedb9eee29942b9721528
